src/main/java/formflow/library/data/SubmissionEncryptionService.java

```diff
@@ -0,0 +1,115 @@
+package formflow.library.data;
+
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.CleartextKeysetHandle;
+import com.google.crypto.tink.JsonKeysetReader;
+import com.google.crypto.tink.aead.AeadConfig;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class SubmissionEncryptionService {
+  private final Aead encDec;
+
+  @Value("${form-flow.inputs: 'org.formflowstartertemplate.app.inputs'}")
+  private String inputConfigPath;
+
+  public SubmissionEncryptionService(@Value("${form-flow.encryption-key:}") String key) {
+    try {
+      AeadConfig.register();
+      encDec = CleartextKeysetHandle.read(JsonKeysetReader.withString(key))
+        .getPrimitive(Aead.class);
+    } catch (GeneralSecurityException | IOException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /**
+   * Go through "Encrypt" annotations and replace fields with encrypted value
+   *
+   * @param submission submission to be encrypted
+   * @return
+   */
+  public Submission encrypt(Submission submission) {
+    if (submission.getFlow() == null) {
+      return submission;
+    }
+
+    try {
+      Class<?> flowClass = Class.forName(inputConfigPath + StringUtils.capitalize(submission.getFlow()));
+      Map<String, Object> inputData = submission.getInputData();
+
+      List<Field> fields = Arrays.stream(flowClass.getDeclaredFields()).filter(field -> Arrays.stream(field.getAnnotations()).anyMatch(annotation -> annotation.annotationType().getName().endsWith("Encrypted"))).toList();
```

Comment:
Is the annotation name a complete class path and so we only care what it ends with?  I wonder if we want (dot) Encrypted (`.Encrypted`) instead, so we can be sure it's the "Encrypted" class and not "SomeOtherMeaningEncrypted"?